src/Components/PayNowButton.res

```diff
@@ -14,41 +14,79 @@ module Loader = {
   }
 }
 @react.component
-let make = () => {
-  let {themeObj, localeString} = Recoil.useRecoilValueFromAtom(RecoilAtoms.configAtom)
+let make = (~cvcProps, ~cardProps, ~expiryProps) => {
+  open RecoilAtoms
+  let {localeString} = Recoil.useRecoilValueFromAtom(configAtom)
   let (isDisabled, setIsDisabled) = React.useState(() => false)
   let (showLoader, setShowLoader) = React.useState(() => false)
-  let complete = Recoil.useRecoilValueFromAtom(RecoilAtoms.fieldsComplete)
+  let areRequiredFieldsValidValue = Recoil.useRecoilValueFromAtom(areRequiredFieldsValid)
+  let {sdkHandleConfirmPayment} = Recoil.useRecoilValueFromAtom(optionAtom)
 
-  let handleOnClick = _ev => {
+  let (isCVCValid, _, _, _, _, _, _, _, _, _) = cvcProps
+  let (isCardValid, _, _, _, _, _, _, _, _, _) = cardProps
+  let (isExpiryValid, _, _, _, _, _, _, _, _) = expiryProps
+
+  let validFormat =
+    isCVCValid->Option.getOr(false) &&
+    isCardValid->Option.getOr(false) &&
+    isExpiryValid->Option.getOr(false) &&
+    areRequiredFieldsValidValue
+
+  let buttonText =
+    sdkHandleConfirmPayment.buttonText->String.length > 0
```

Comment:
Instead of checking this on length, could you please keep the check as

`sdkHandleConfirmPayment.buttonText !== ""`